bson/__init__.py

```diff
@@ -388,6 +392,12 @@ def _get_binary(
     # Decode subtype 0 to 'bytes'.
     if subtype == 0:
         value = data[position:end]
+    # Decode subtype 9 to vector.
+    elif subtype == 9:
+        dtype, padding = struct.unpack_from("<sB", data, position)
+        position += 2
+        end = position + length
+        value = BinaryVector(data[position:end], dtype=dtype, padding=padding)
```

Comment:
We shouldn't have any custom vector encoding or decoding logic here. vector subtypes  should be decoded to regular bson.Binary instances the same as any other subtype.

Reply:
I disagree. The Binary class needs to contain the actual bson data otherwise we're breaking the Liskov Substitution Principle. We could always extend the api with a new Vector by adding a new decoding setting like codec_options.vector_conversion`.

Reply:
Ah, good call out.  Yeah, I like the idea of `codec_options.vector_conversion`.

Reply:
I'm not against adding a vector_conversion but I don't think we should do it in this project. Right now we don't have any real use cases so we should stick with the simple Binary.from_list/as_list apis. Once we have more users we can tackle extending the design via vector_conversion. What do you think?

Reply:
Yeah, in the interest of getting this into the 4.9 release I think de-scoping it make sense.